GoogleContainerTools/skaffold
# feat: Add privateWorkerPool and location configuration for gcb

Fixes: #7342

## Description
 - cherry-pick privateWorkerPool support from main
 - cherry-pick region configuration in googleCouldBuild support from main
 - implement change to support configure the region, if workerPool is configured, build location is still deduced from workerPool. If not and region is configured, region in the configuration will be used as build location.

## User facing changes

 - Add privateWorkerPool and region configuration for gcb, so users are able to configure region to run gcb

## Test Plan
 - change [this line](https://github.com/ericzzzzzzz/skaffold/blob/add-location-configuration-for-gcb/integration/run_test.go#L241) to ``MarkIntegrationTest(t, CanRunWithoutGcp)`` then manually run [this test](https://github.com/ericzzzzzzz/skaffold/blob/add-location-configuration-for-gcb/integration/run_test.go#L260-L264)
 - enter gcp console, select k8s-skaffold project, then check the build history from asia-east1, should be able to find the build detail.
